library/general/src/lib/yast2/log_dir_rotator.rb

```diff
@@ -0,0 +1,195 @@
+# encoding: utf-8
+
+# ***************************************************************************
+#
+# Copyright (c) 2018-2019 SUSE LLC
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of version 2 of the GNU General Public License as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.   See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, contact Novell, Inc.
+#
+# To contact Novell about this file by physical or electronic mail,
+# you may find current contact information at www.suse.com.
+#
+# ***************************************************************************
+
+require "yast"
+require "fileutils"
+
+Yast.import "Mode"
+Yast.import "Directory"
+
+module Yast2
+  # The class LogDirRotator handels creating, rotating and removing
+  # directories for logging in the YaST2 log directory according to
+  # this schema:
+  #
+  #   rm #{NAME}-03
+  #   mv #{NAME}-02 #{NAME}-03
+  #   mv #{NAME}-01 #{NAME}-02
+  #   mv #{NAME}    #{NAME}-01
+  #
+  # System error are logged and no exception is raised by the functions.
+  #
+  # TODO This class is based on dump_manager.rb and could be used there.
+  class LogDirRotator
+    include Yast::Logger
+
+    # The number of old log directories to keep. This is in addition
+    # to the current one. Child classes can override the value. Max
+    # value is 99.
+    KEEP_OLD_LOG_DIRS = 3
+
+    # The name of the directory in /var/log/YaST2 to create, rotate
+    # and delete. Child classes should override the value.
+    NAME = "undefined".freeze
+
+    # Prepare (create, rotate and delete) the log directory.
+    def prepare
+      return if Yast::Mode.test
+      begin
+        log.info "preparing log dir #{self.class::NAME}"
+        rotate_log_dirs
+        FileUtils.mkdir_p(log_dir)
+      rescue SystemCallError => e
+        log.error e.to_s
+      end
+    end
+
+    # Copy a file to the log directory.
+    #
+    # @param src [String] name of source
+    # @param dest [String] basename of destination
+    def copy(src, dest)
+      return if Yast::Mode.test
+      begin
+        FileUtils.cp(src, log_dir + "/" + dest)
+      rescue SystemCallError => e
+        log.error e.to_s
+      end
+    end
+
+    # Write a file in the log directory.
+    #
+    # @param dest [String] basename of destination
+    # @param content [String] content to write
+    def write(dest, content)
+      return if Yast::Mode.test
+      begin
+        File.open(log_dir + "/" + dest, "w") { |file| file.write(content) }
```

Comment:
It can be written as `File.write(log_dir + "/" + dest, content)` https://ruby-doc.org/core-2.5.0/IO.html#method-c-write